packages/server-side-render/src/server-side-render.js

```diff
@@ -20,126 +20,105 @@ export function rendererPath( block, attributes = null, urlQueryArgs = {} ) {
 	} );
 }
 
-export class ServerSideRender extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			response: null,
-		};
-	}
+export function ServerSideRender( props ) {
+	const {
+		className,
+		EmptyResponsePlaceholder,
+		ErrorResponsePlaceholder,
+		LoadingResponsePlaceholder,
+	} = props;
 
-	componentDidMount() {
-		this.isStillMounted = true;
-		this.fetch( this.props );
-		// Only debounce once the initial fetch occurs to ensure that the first
-		// renders show data as soon as possible.
-		this.fetch = debounce( this.fetch, 500 );
-	}
+	const [ response, setResponse ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const isMounted = useRef( true );
 
-	componentWillUnmount() {
-		this.isStillMounted = false;
-	}
+	useEffect( () => {
+		fetchData( props );
 
-	componentDidUpdate( prevProps ) {
-		if ( ! isEqual( prevProps, this.props ) ) {
-			this.fetch( this.props );
-		}
-	}
+		fetchData = debounce( fetchData, 500 );
+		return () => {
+			isMounted.current = false;
+		};
+	}, [] );
 
-	fetch( props ) {
-		if ( ! this.isStillMounted ) {
+	useEffect( () => {
+		fetchData( props );
+	}, [ props ] );
+
+	function fetchData( options ) {
+		if ( ! isMounted ) {
 			return;
 		}
-		if ( null !== this.state.response ) {
-			this.setState( { response: null } );
-		}
+		setIsLoading( true );
 		const {
 			block,
 			attributes = null,
 			httpMethod = 'GET',
 			urlQueryArgs = {},
-		} = props;
+		} = options;
 
 		// If httpMethod is 'POST', send the attributes in the request body instead of the URL.
 		// This allows sending a larger attributes object than in a GET request, where the attributes are in the URL.
 		const isPostRequest = 'POST' === httpMethod;
 		const urlAttributes = isPostRequest ? null : attributes;
 		const path = rendererPath( block, urlAttributes, urlQueryArgs );
 		const data = isPostRequest ? { attributes } : null;
+		let currentFetchRequest = null;
```

Comment:
This needs to be a ref at the component level, not a variable local to the current instance of the `fetchData` callback.

Reply:
addressed  👍